pkg/grafana/grafana.go

```diff
@@ -102,6 +139,123 @@ func DeleteByID(ctx context.Context, grafanaAPI *client.GrafanaHTTPAPI, id int64
 	return nil
 }
 
+func ConfigureDefaultDatasources(ctx context.Context, grafanaAPI *client.GrafanaHTTPAPI, organization Organization) ([]Datasource, error) {
+	logger := log.FromContext(ctx)
+
+	// TODO using a serviceaccount later would be better as they are scoped to an organization
+
+	// Switch context to the current org
+	if _, err := grafanaAPI.SignedInUser.UserSetUsingOrg(organization.ID); err != nil {
+		logger.Error(err, "failed to change current org for signed in user")
+		return nil, errors.WithStack(err)
+	}
+
+	// We always switch back to the shared org
+	defer func() {
+		if _, err := grafanaAPI.SignedInUser.UserSetUsingOrg(SharedOrg.ID); err != nil {
```

Comment:
Why? The err is not defined and it's scoped to the if

Reply:
`err` is already defined in the parent function, therefore this defer function can and should assign the error to it in order to return the error if any from the parent function.

Reply:
It's not defined in the parent function, it's scoped to the previous if. Also, this err is done as part of a defer function so even if it would set the value of err, it should not bé able to bé returned from the parent function right?

Reply:
yes it would be returned by the parent function, see https://go.dev/play/p/20q8qQHZPn_m

Please define the err in the parent function and assign it as describe here.

Reply:
You can also make it a naked return right? Alright i'll do that on monday